### PR TITLE
Clear all `yarn lint-ts` warnings

### PR DESCRIPTION
**Pull request by julieg18**
* updates our `.eslintrc.json` `overrides.files` array, adding folders that have `js` files. Otherwise we get typescript warnings on some our `js` files in `src`.
* fixes typescript errors in `src/components/organisms/SiteHeader`

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -36,7 +36,9 @@
       "files": [
         "src/utils/shared/*.?(js|ts)",
         "src/gatsby/**/*.js",
-        "src/components/PageWrapper/index.js",
+        "src/components/**/*.js",
```
Comment (julieg18) [inline .eslintrc.json:39]: I just added the folders that includes `.js` files which is what dvc.org does, but is there a reason we can't just add `src/**/*.js` to this array?

Comment (casperdcl) [reply .eslintrc.json:39]: I'd prefer the catch-all `**` too...

Comment (rogermparent) [reply .eslintrc.json:39]: `**` should be fine, if we really need to ignore something we can add it to the ignore list anyway.

**Review by rogermparent — APPROVED** [review]

File: .eslintrc.json [hunk 2]
```diff
@@ -34,12 +34,9 @@
   "overrides": [
     {
       "files": [
-        "src/utils/shared/*.?(js|ts)",
-        "src/gatsby/**/*.js",
-        "src/components/PageWrapper/index.js",
+        "src/**/*.js",
         "scripts/**/*.js",
         "config/**/*.js",
```
Comment (casperdcl) [inline .eslintrc.json:39]: what about this?

```suggestion
        "**/*.js"
```

File: .eslintrc.json [hunk 3]
```diff
@@ -34,12 +34,9 @@
...
-        "src/server/**/*.js",
         "gatsby-*.js",
         "postcss.config.js"
```
Comment (casperdcl) [inline .eslintrc.json:41]:
```suggestion
```

**Review by casperdcl — COMMENTED** [review]
can we simplify fürther?

**Pull request merged**